demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java

```diff
@@ -66,24 +72,72 @@ public void run() {
         List<CompletedDownloadBatch> completeDownloadBatches = migrationExtractor.extractMigrations();
 
         onUpdate("Queuing Partial Downloads");
-        for (VersionOnePartialDownloadBatch partialDownloadBatch : partialDownloadBatches) {
+        for (int i = partialDownloadBatches.size() - 1; i >= 0; i--) {
+            VersionOnePartialDownloadBatch partialDownloadBatch = partialDownloadBatches.get(i);
             downloadManager.download(partialDownloadBatch.batch());
+            partialDownloadBatches.remove(i);
 
             for (String originalFileLocation : partialDownloadBatch.originalFileLocations()) {
-                deleteVersionOneFile(originalFileLocation);
+                boolean usedInFurtherBatches = originalFileUsedInFurtherBatches(
+                        originalFileLocation,
+                        partialDownloadBatches,
+                        completeDownloadBatches
+                );
+                if (!usedInFurtherBatches) {
+                    deleteVersionOneFile(originalFileLocation);
+                }
             }
         }
 
         onUpdate("Migrating Complete Downloads");
-        for (CompletedDownloadBatch completeDownloadBatch : completeDownloadBatches) {
-            downloadManager.addCompletedBatch(completeDownloadBatch);
+
+        for (int i = completeDownloadBatches.size() - 1; i >= 0; i--) {
+            CompletedDownloadBatch completedDownloadBatch = completeDownloadBatches.get(i);
+            downloadManager.addCompletedBatch(completedDownloadBatch);
+            completeDownloadBatches.remove(i);
+
+            for (CompletedDownloadFile completedDownloadFile : completedDownloadBatch.completedDownloadFiles()) {
+                migrateV1FileToV2Location(completedDownloadFile);
+
+                boolean usedInFurtherBatches = originalFileUsedInFurtherBatches(
+                        completedDownloadFile.originalFileLocation(),
+                        partialDownloadBatches,
```

Comment:
At this point `partialDownloadBatches` is empty, right?

Reply:
Yea should be, I didn't want to create another method that did the same thing but had one list less 😂